### ISSUE TITLE
permit setting all vType parameters

### ISSUE THREAD

**Issue by namdre** [eclipse-sumo/sumo#10430]
there are a number of vType parameters (i.e. jmIgnoreFoeProb) that cannot currently by manipulated 

(see #10427)

define new traci constants and functions:

- [ ] collisionMinGapFactor
- [ ] imgFile
- [ ] carFollowModel
- [ ] personCapacity (cannot be set at the moment)
- [ ] containerCapacity
- [x] boardingDuration
- [ ] loadingDuration
- [x] scale
- [x] impatience
- [x] jmCrossingGap
- [x] jmIgnoreKeepClearTime
- [x] jmDriveAfterRedTime
- [x] jmDriveAfterYellowTime
- [x] jmDriveRedSpeed
- [x] jmIgnoreFoeProb
- [x] jmIgnoreFoeSpeed
- [x] jmIgnoreJunctionFoeProb
- [x] jmSigmaMinor
- [x] jmStoplineGap
- [x] jmTimegapMinor
- [ ] startupDelay (#7832)
- [ ] sigmaStep
- [ ] timeToTeleport
- [ ] timeToTeleportBidi
- [ ] desiredMaxSpeed

**Comment by behrisch** [comment]
- first check whether we have enough constants left because they also get passed to the vehicle
- differentiate between first class and second class parameters (let jm* be second class as well as future cf model params EIDM etc.)

**Comment by rummel123** [comment]
I need to get the `boardingDuration` via traci. Maybe its possible to add the `getBoardingDuration` method provisionaly?

**Comment by namdre** [comment]
since you cannot change it via TraCI, why not parse it directly out of the simulation input files?

**Comment by rummel123** [comment]
I need it for the drtOrtools.py and I don't want to parse the (unknown) simulation files.

**Comment by namdre** [comment]
@behrisch The constant 0x3f was used by @m-kro and then you changed it in 28c0b5f919a621ca846b9660ee7e0a81e6e846d3

It would probably make sense to add it as a placeholder (i.e. `TRACI_CONST int RESERVED_FOR_XYZ_DO_NOT_USE = 0x3f`)

**Comment by behrisch** [comment]
There is nothing wrong with using 0x3f it is just wasteful because we may want to keep it for a parameter applicable to "every" domain

**Comment by MustaphaBichi** [comment]
Sorry to ask you Namdre, can MinGap parameter be changed? if yes, Is it possible to set the new value in route definitions?

**Comment by namdre** [comment]
yes: `traci.vehicletype.setMinGap` and `traci.vehicle.setMinGap`.  
If you want to load this from xml, you have to define it as `<vType ... minGap="VALUE"/>`

**Comment by MustaphaBichi** [comment]
Ok, Thank you for your response.

**Comment by MustaphaBichi** [comment]
> yes: `traci.vehicletype.setMinGap` and `traci.vehicle.setMinGap`. If you want to load this from xml, you have to define it as `<vType ... minGap="VALUE"/>`

Thank you for your kind response.

**Comment by MustaphaBichi** [comment]
> yes: `traci.vehicletype.setMinGap` and `traci.vehicle.setMinGap`. If you want to load this from xml, you have to define it as `<vType ... minGap="VALUE"/>`

I created additional file as seen below, in order to define two vehicle type; passenger, and bus, with minGap of 2m.
<additional>
	<vType id="car" minGap="2" length="5" vClass="passenger"/>
	<vType id="bus" minGap="2" length="12" vClass="bus"/>
</additional>

I loaded the type file in my configuration file with the code below. after running the simulation, there is no effect i.e only passenger car were shown. 
<configuration>
	<input>
		<net-file value="sample.net.xml"/>
		<additional-files value="type.add.xml"/>
		<route-files value="sample.rou.xml"/>
	</input>

Secondly, I created additional file as seen below, with the file extension add.xml
<routes>
	<vType id="car" minGap="2" length="5" vClass="passenger"/>
	<vType id="bus" minGap="2" length="12" vClass="bus"/>
</routes>
I loaded the file while generating randomTrips, as seen in the code below, but only the defult setting was used (i.e pessenger car)

C:\Sumo\My Sumo Work\sample>python randomTrips.py -n sample.net.xml --additional-file additional.add.xml -r sample.rou.xml -e 1000 -p 1 --validate
Please.. can you help me out? I really want to create a network using randomTrip with multiple vehicle type (passenger, bus, and trunk), with minGap of 1m.

**Comment by MustaphaBichi** [comment]
I created additional file as seen below, in order to define two vehicle type; passenger, and bus, with minGap of 2m.
`<additional>
	<vType id="car" minGap="2" length="5" vClass="passenger"/>
	<vType id="bus" minGap="2" length="12" vClass="bus"/>
</additional>
`
I loaded the type file in my configuration file with the code below. after running the simulation, there is no effect i.e only passenger car were shown.
`I loaded the type file in my configuration file with the code below. After running the simulation, there is no effect i.e only passenger car were shown.`

Secondly, I created additional file as seen below, with the file extension add.xml
`<routes>
	<vType id="car" minGap="2" length="5" vClass="passenger"/>
	<vType id="bus" minGap="2" length="12" vClass="bus"/>
</routes>`

I loaded the file while generating randomTrips, as seen in the code below, but only the defult setting was used (i.e pessenger car)
`
C:\Users\musty\Sumo\My Sumo Work\sample>python randomTrips.py -n sample.net.xml --additional-file additional.add.xml -r sample.rou.xml -e 1000 -p 1 --validate`

Please.. can you help me out? I really want to create a network using randomTrip with multiple vehicle type (passenger, bus, and trunk), with minGap of 1m.

**Comment by namdre** [comment]
see https://sumo.dlr.de/docs/Tools/Trip.html#setting_a_vehicle_type_from_an_external_file (but please don't post questions that are unrelated to the original ticket, rather open a new ticket)

**Comment by MustaphaBichi** [comment]
Noted, Thank you
